pkgs/games/darkplaces/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, zlib
+, libjpeg
+, SDL2
+, libvorbis
+}:
+stdenv.mkDerivation rec {
+  name = "darkplaces";
+  version = "unstable-2022-05-10";
+  src = fetchFromGitHub {
+    owner = "DarkPlacesEngine";
+    repo = "darkplaces";
+    rev = "f16954a9d40168253ac5d9890dabcf7dbd266cd9";
+    hash = "sha256-5KsUcgHbuzFUE6LcclqI8VPSFbXZzBnxzOBB9Kf8krI=";
+  };
+
+  buildInputs = [
+    zlib
+    libjpeg
+    SDL2
+  ];
+
+  buildFlags = "release";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    install -m755 darkplaces-sdl $out/bin/darkplaces
+    install -m755 darkplaces-dedicated $out/bin/darkplaces-dedicated
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    patchelf \
+      --add-needed ${libvorbis}/lib/libvorbisfile.so \
+      --add-needed ${libvorbis}/lib/libvorbis.so \
+      $out/bin/darkplaces
```

Comment:
```suggestion
    patchelf \
      --add-needed libvorbisfile.so \
      --add-needed libvorbis.so \
      $out/bin/darkplaces
```

Reply:
@SuperSandro2000 this doesn't really seem to work. I think it needs the full package path? Maybe I'm just doing something weird, but it doesn't appear to work.

Reply:
Additionally, if I do NOT patch it in using this method and try to just add it as a build dependency, it doesn't seem to properly include the libvorbis and libvorbisfile support, and no .ogg music files will play.

Reply:
If the package is added to buildInputs it shouldn't require that.

Reply:
I agree, but when I'm building it, it doesn't appear to properly link against the libvorbis libraries.

Reply:
I'm not disagreeing with you on that.

```
  postFixup = ''
    patchelf \
      --add-needed libvorbisfile.so \
      $out/bin/darkplaces
  '';
```

Doing the above, results in the following error:

```
/nix/store/nx41mlpg71lfpznjf495fxk3amghsxp2-darkplaces/bin/darkplaces: error while loading shared libraries: libvorbisfile.so: cannot open shared object file: No such file or directory
```

when trying to execute the application. that's why I set it to the full path.

Reply:
🤔 thats strange. Then we stick to what works.